torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -189,11 +189,25 @@ void IrPrinter::handle(const kir::Half* h) {
 }
 
 void IrPrinter::handle(const kir::Int* i) {
-  os_ << "kir::Int";
```

Comment:
All of the kir nodes printing will go away from IrPrinter shortly. If you're ok with that then the changes seem fine, but they will be short lived.

Reply:
Is it possible to modify the expression evaluator to handle comparing symbolic equations? e.g. check if two sizes are equal at compile time

Reply:
Do you mean, check if they are equal w/o actually evaluating them? What is the use case you have in mind?

Reply:
One alias check is to see if the size of the allocations are equal, so I want to check if their symbolic expressions are equal. Currently, I am using the IrPrinter for this purpose.

Reply:
Interesting. 

I'd recommend against using IrPrinter for this purpose - primarily because it's not intended for this use case so it's not guaranteed to be reliable. I'm willing to bet that it's also terribly slow. Finally, it only checks for identical expressions, which is not the same as symbolic equivalence (a simple variable renaming would throw off the whole check)

I understand that this is the most immediate solution, and it may be ok as a temporary hack but we should add some TODOs and open a Github issue to revisit this (and add support for a more reliable expression equivalence check)